cmd/dagger/check.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/dagger/dagger/engine/client"
+	"github.com/iancoleman/strcase"
+	"github.com/juju/ansiterm/tabwriter"
+	"github.com/moby/buildkit/identity"
+	"github.com/muesli/termenv"
+	"github.com/opencontainers/go-digest"
+	"github.com/spf13/cobra"
+	"github.com/vito/progrock"
+	"golang.org/x/sync/errgroup"
+)
+
+var checkCmd = &cobra.Command{
+	Use:                "checks [suite]",
+	DisableFlagParsing: true,
+	Long:               `Query the status of your environment's checks.`,
+	RunE:               loadEnvCmdWrapper(RunCheck),
+}
+
+var (
+	doFocus bool
+)
+
+func init() {
+	rootCmd.AddCommand(checkCmd)
+
+	checkCmd.PersistentFlags().BoolVar(&doFocus, "focus", true, "Only show output for focused commands.")
+
+	checkCmd.AddCommand(
+		&cobra.Command{
+			Use:          "list",
+			Long:         `List your environment's checks without updating their status.`,
+			SilenceUsage: true,
+			RunE:         loadEnvCmdWrapper(ListChecks),
+		},
+	)
+
+}
+
+func ListChecks(ctx context.Context, _ *client.Client, c *dagger.Client, env *dagger.Environment, cmd *cobra.Command, dynamicCmdArgs []string) (err error) {
+	rec := progrock.RecorderFromContext(ctx)
+	vtx := rec.Vertex("cmd-list-checks", "list checks", progrock.Focused())
+	defer func() { vtx.Done(err) }()
+
+	envChecks, err := env.Checks(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get environment commands: %w", err)
+	}
+
+	tw := tabwriter.NewWriter(vtx.Stdout(), 0, 0, 2, ' ', 0)
+
+	if stdoutIsTTY {
+		fmt.Fprintf(tw, "%s\t%s\n", termenv.String("check name").Bold(), termenv.String("description").Bold())
+	}
+
+	var printCheck func(*dagger.Check) error
+	printCheck = func(check *dagger.Check) error {
+		name, err := check.Name(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get check name: %w", err)
+		}
+		name = strcase.ToKebab(name)
+
+		descr, err := check.Description(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get check description: %w", err)
+		}
+		fmt.Fprintf(tw, "%s\t%s\n", name, descr)
+		return nil
+	}
+
+	for _, check := range envChecks {
+		// TODO: this shouldn't be needed, there is a bug in our codegen for lists of objects. It should
+		// internally be doing this so it's not needed explicitly
+		checkID, err := check.ID(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get check id: %w", err)
+		}
+
+		check = *c.Check(dagger.CheckOpts{ID: checkID})
+		err = printCheck(&check)
+		if err != nil {
+			return err
+		}
+	}
+
+	return tw.Flush()
+}
+
+func RunCheck(ctx context.Context, _ *client.Client, c *dagger.Client, env *dagger.Environment, cmd *cobra.Command, dynamicCmdArgs []string) (err error) {
+	rec := progrock.RecorderFromContext(ctx)
+
+	// TODO(vito): this is pretty confusing, but we need to initialize a root
+	// group for subsequent groups + vertices
+	rec = rec.WithGroup(progrock.RootGroup)
```

Comment:
This can be removed after #5681 is merged (details: ["Single root invariant"](https://github.com/vito/progrock/pull/11))